File: Backend/src/Meseta_Verda.Domain/Entities/Repartidor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

namespace Meseta_Verda.Domain.Entities
{
    public class Repartidor
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int IdUsuario { get; set; }
        [ForeignKey("IdUsuario")]
        public Usuario Usuario { get; set; } 
        public string PlacaVehiculo { get; set; } = string.Empty;
        public string Estado { get; set; } = "DISPONIBLE"; // "EN ENTREGA", "DISPONIBLE", "NO DISPONIBLE"
        public string Vehiculo { get; set; } = string.Empty;
        public decimal PromedioCalificacion { get; set; }
        public string CuentaBancaria { get; set; } = string.Empty;
        public string UrlFotoPerfil { get; set; } = string.Empty;
        public string ZonaOperaciones { get; set; } = string.Empty;
        public string Departamento { get; set;} = string.Empty;



    }
}

File: Frontend/agrotrade_frontend/lib/models/Repartidor/perfil/repartidor.dart
class Repartidor {
  final int id;
  final int idUsuario;
  final String placaVehiculo;
  final String estado;
  final String vehiculo;
  final double promedioCalificacion;
  final String cuentaBancaria;
  final String urlFotoPerfil;
  final String zonaOperaciones;
  final String departamento;

  const Repartidor({
    required this.id,
    required this.idUsuario,
    required this.placaVehiculo,
    this.estado = "DISPONIBLE",
    required this.vehiculo,
    required this.promedioCalificacion,
    required this.cuentaBancaria,
    required this.urlFotoPerfil,
    required this.zonaOperaciones,
    required this.departamento,
  });

  factory Repartidor.fromJson(Map<String, dynamic> json) {
    return Repartidor(
      id: json['id'] as int,
      idUsuario: json['idUsuario'] as int,
      placaVehiculo: json['placaVehiculo'] as String? ?? '',
      estado: json['estado'] as String? ?? 'DISPONIBLE',
      vehiculo: json['vehiculo'] as String? ?? '',
      promedioCalificacion: (json['promedioCalificacion'] as num?)?.toDouble() ?? 0.0,
      cuentaBancaria: json['cuentaBancaria'] as String? ?? '',
      urlFotoPerfil: json['urlFotoPerfil'] as String? ?? '',
      zonaOperaciones: json['zonaOperaciones'] as String? ?? '',
      departamento: json['departamento'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'idUsuario': idUsuario,
      'placaVehiculo': placaVehiculo,
      'estado': estado,
      'vehiculo': vehiculo,
      'promedioCalificacion': promedioCalificacion,
      'cuentaBancaria': cuentaBancaria,
      'urlFotoPerfil': urlFotoPerfil,
      'zonaOperaciones': zonaOperaciones,
      'departamento': departamento,
    };
  }
}
